src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/cdk/cdk_stacks/adf_standard_pipelines.py

```diff
@@ -0,0 +1,260 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+"""This is the functionality for generating a default adf pipeline.
+"""
+
+import os
+
+from aws_cdk import aws_codepipeline as _codepipeline, core
+from cdk_constructs import adf_codepipeline
+from cdk_constructs import adf_codebuild
+from cdk_constructs import adf_jenkins
+from cdk_constructs import adf_codecommit
+from cdk_constructs import adf_github
+from cdk_constructs import adf_s3
+from cdk_constructs import adf_cloudformation
+from cdk_constructs import adf_notifications
+from logger import configure_logger
+
+ADF_DEPLOYMENT_REGION = os.environ["AWS_REGION"]
+ADF_DEPLOYMENT_ACCOUNT_ID = os.environ["ACCOUNT_ID"]
+ADF_DEFAULT_BUILD_TIMEOUT = 20
+LOGGER = configure_logger(__name__)
+
+PIPELINE_TYPE = "default"
+
+
+def generate_standard_adf_pipeline(scope: core.Stack, stack_input):
+    _stages = []
+
+    if stack_input["input"].get("params", {}).get("notification_endpoint"):
+        stack_input["input"]["topic_arn"] = adf_notifications.Notifications(
+            scope, "adf_notifications", stack_input["input"]
+        ).topic_arn
+
+    _source_name = generate_source_stage_for_pipeline(_stages, scope, stack_input)
+    generate_build_stage_for_pipeline(_stages, scope, stack_input)
+    print(stack_input["input"].get("environments", {}).get("targets", []))
+    for index, targets in enumerate(stack_input["input"].get("environments", {}).get("targets", [])):
```

Comment:
Would it be easy to move the target generation to its own function too?
In that case this function will become nothing else than setting up the notification endpoint, rendering source, build, and targets. Plus generating the related Pipeline at the end + webhooks for Github.